pennylane/ops/op_math/controlled.py

```diff
@@ -556,7 +556,12 @@ def has_generator(self):
 
     def generator(self):
         sub_gen = self.base.generator()
-        proj_tensor = operation.Tensor(*(qml.Projector([1], wires=w) for w in self.control_wires))
+        proj_tensor = operation.Tensor(
+            *(
+                qml.Projector([val], wires=w)
+                for val, w in zip(self.control_values, self.control_wires)
+            )
+        )
```

Comment:
While we're changing this line...

We can return `qml.operation.active_new_opmath()`, return ` qml.prod` if so, and return `Tensor` otherwise.  This will also help us in our transition to deprecating `Tensor`.

Reply:
definitely. might wanna check out #4578 before making changes like that though - it has burned us before